### PR TITLE
Move handle distribution explanation to freshness section

### PR DESCRIPTION
**Pull request by dthaler**
Per last week's design team discussion.

Signed-off-by: Dave Thaler <dthaler@ntdev.microsoft.com>

### CODE REVIEW

**Review by William-PanWei — APPROVED** [review]

**Review by thomas-fossati — CHANGES_REQUESTED** [review]

File: draft-ietf-rats-architecture.md [hunk 1]
```diff
@@ -1095,16 +1095,45 @@ Attestation Result.  In this case the epoch is said to be rough because:
...
 
+## Implicit Timekeeping using Epoch Handles
+
+A third approach relies on having epoch "handles"
+periodically sent to both the sender and receiver of Evidence or
+Attestation Results by some "Handle Distributor".
+
+Handles are different from nonces as they can be used more than once and
+can even be used by more than one entity at the same time.
+Handles are different from timestamps as they do not have to convey information about a point in time.
```
Comment (mcr) [inline draft-ietf-rats-architecture.md:1106]:
```suggestion
Handles are different from timestamps as they do not have to convey information about a point in time, i.e., they are not necessarily monotonically increasing integers.
```

File: draft-ietf-rats-architecture.md [hunk 2]
```diff
@@ -1095,16 +1095,45 @@ Attestation Result.  In this case the epoch is said to be rough because:
...
+
+The most recent handle is included in the produced Evidence or Attestation
+Results, and the appraising entity can compare the handle in received
+Evidence or Attestation Results against latest handle it received from
+the Handle Distributor to determine if it is within the current epoch.
+An actual solution also needs to take into account race conditions
+when transitioning to a new epoch, such as by using an epoch counter
+in addition to an epoch handle, or by including both the current and
+previous handles in messages and/or checks, or by requiring retries
+in case of mismatching handles.
```
[the remark this answers is not in the document]
Comment (mcr) [reply draft-ietf-rats-architecture.md:1122]: "obtained" or "received"?

File: draft-ietf-rats-architecture.md [hunk 3]
```diff
@@ -1095,16 +1095,45 @@ Attestation Result.  In this case the epoch is said to be rough because:
...
+appraise the freshness of Evidence or Attestation Results.  Only the
+Handle Distributor requires access to a clock so it can periodically send
+new epoch handles.
+
+The most recent handle is included in the produced Evidence or Attestation
+Results, and the appraising entity can compare the handle in received
+Evidence or Attestation Results against latest handle it received from
+the Handle Distributor to determine if it is within the current epoch.
+An actual solution also needs to take into account race conditions
+when transitioning to a new epoch, such as by using an epoch counter
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:1119]: An epoch counter is (sometimes persistent) state on an entity receiving handles that basically retains a n handles in, for example, a ring buffer and therefore provides a record of the past n handles in sequence.

File: draft-ietf-rats-architecture.md [hunk 4]
```diff
@@ -1095,16 +1095,45 @@ Attestation Result.  In this case the epoch is said to be rough because:
...
+in addition to an epoch handle, or by including both the current and
+previous handles in messages and/or checks, or by requiring retries
+in case of mismatching handles.
+
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-architecture.md:1123]: While non-trivial to capture by a single entity involved, this is crucial to understand the overlap phase "in a bigger picture" and how to conduct appraisal procedures later on, I think.

**Comment by henkbirkholz** [comment]
In general, I like the re-introduction of the "timekeeping" subsections a lot.

+1

File: draft-ietf-rats-architecture.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dthaler) [reply draft-ietf-rats-architecture.md:1123]: Disagree, the prop delay is irrelevant in all of my text, and no estimation is needed.

File: draft-ietf-rats-architecture.md (hunk 1 above)
Comment (dthaler) [reply draft-ietf-rats-architecture.md:1106]: Ok

File: draft-ietf-rats-architecture.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dthaler) [reply draft-ietf-rats-architecture.md:1122]: Obtained sounds better ("receiver received" sounds odd to me).  Thomas's text looks ok to me.

File: draft-ietf-rats-architecture.md [hunk 5]
```diff
@@ -1095,16 +1095,45 @@ Attestation Result.  In this case the epoch is said to be rough because:
...
+Handle Distributor requires access to a clock so it can periodically send
+new epoch handles.
+
+The most recent handle is included in the produced Evidence or Attestation
+Results, and the appraising entity can compare the handle in received
+Evidence or Attestation Results against the latest handle it received from
+the Handle Distributor to determine if it is within the current epoch.
+An actual solution also needs to take into account race conditions
+when transitioning to a new epoch, such as by using a counter signed
+by the Handle Distributor as the handle, or by including both the current and
```
Comment (dthaler) [inline draft-ietf-rats-architecture.md:1120]: Updated text per Thomas's comment.   If the handle is a signed counter, and the handle is sent in (say) evidence, then the verifier receiving it that hasn't seen the new handle yet can simply accept it if it's signed by the HD it trusts and it sees the counter is higher than the handle it has.

**Pull request merged**